modules/build/src/main/scala/scala/build/Build.scala

```diff
@@ -314,15 +314,26 @@ object Build {
     classesRootDir(root, projectName) / scope.name
 
   def scalaNativeSupported(options: BuildOptions, inputs: Inputs) = either {
-    val version = value(options.scalaParams).scalaVersion
-    if (version.startsWith("3")) false
-    else if (version.startsWith("2.13")) Properties.isMac || Properties.isLinux
-    else if (version.startsWith("2.12"))
-      !inputs.sourceFiles().exists {
-        case _: Inputs.AnyScript => true
-        case _                   => false
-      }
-    else false
+    val scalaVersion  = value(options.scalaParams).scalaVersion
+    val nativeVersion = options.scalaNativeOptions.numeralVersion
+    nativeVersion match {
+      case Some(snNumeralVer) =>
+        if (snNumeralVer < SNNumeralVersion(0, 4, 1) && Properties.isWin)
+          false
+        else if (scalaVersion.startsWith("3.0"))
+          false
+        else if (scalaVersion.startsWith("3"))
+          snNumeralVer >= SNNumeralVersion(0, 4, 3)
+        else if (scalaVersion.startsWith("2.13"))
+          true
+        else if (scalaVersion.startsWith("2.12"))
+          !inputs.sourceFiles().exists {
+            case _: Inputs.AnyScript => true
+            case _                   => false
+          }
```

Comment:
Also, this looks like a candidate for refactoring, no need for negation if we can simply set it to correct value in the first place, becouse `!exists` can be replaced with `forall`. We can simplify it even further to 
```suggestion
          inputs.sourceFiles().forall(!_.isInstanceOf[Input.AnyScript])
```

Reply:
In a previous PR I saw that using pattern matching instead of isInstanceOf may have benefits here - we'll get a compiler error if the type of sourceFile changes (https://github.com/VirtusLab/scala-cli/pull/178#discussion_r723368690). I will change to forall though